apps/app/src/views/routes/Main.tsx

```diff
@@ -61,7 +64,9 @@ export const Main = () => {
   }
 
   if (!isAuthenticated && !isLoading) {
-    return <Navigate to="/login" state={{ from: location }} replace />;
+    // forwarding on the query param to the login page
+    const to = `/login${query.get('auth0-orgs') === '0' ? '?auth0-orgs=0' : ''}`;
```

Comment:
can we just forward all query params? the case i think would be (and good to test):
- i intend to go to like `/orders/new?patientId=1234`
- i click login
- I select my personal account by mistake
- i see the error message, i click login again
- I select my work email
- i have an org and get direct to the creat eorder screen for that patient

Reply:
I updated this to forward all query params, but we're achieving the "return to" after login in the next line by setting state:
```
return <Navigate to={to} state={{ from: location }} replace />;
```
The issue I'm running into trying to pass along location state after a failed login is that when the user is logged out after signing in to the wrong account, the `returnTo` url has to be set in the auth0 admin. In an ideal scenario, I'd just redirect the user to the route they wanted to go after logout... which would kick them back to login and track the location state.

Reply:
solved this by passing another query param `pathname` along back to the redirect url